extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java

```diff
@@ -150,14 +153,15 @@ public void openshiftRequirementsJvm(OpenshiftConfig openshiftConfig,
                 envProducer.produce(KubernetesEnvBuildItem.createSimpleVar(b.getJvmOptionsEnvVar(),
                         String.join(" ", config.getEffectiveJvmArguments()), null));
             });
+
             //In all other cases its the responsibility of the image to set those up correctly.
-            if (!baseImage.isPresent()) {
+            if (hasCustomJarPath || hasCustomJvmArguments) {
                 List<String> cmd = new ArrayList<>();
                 cmd.add("java");
                 cmd.addAll(config.getEffectiveJvmArguments());
                 cmd.addAll(Arrays.asList("-jar", pathToJar));
                 envProducer.produce(KubernetesEnvBuildItem.createSimpleVar("JAVA_APP_JAR", pathToJar, null));
-                commandProducer.produce(KubernetesCommandBuildItem.command(cmd));
+                commandProducer.produce(new KubernetesCommandBuildItem(cmd, config.jvmArguments.get()));
```

Comment:
This is not correct. One thing is the JVM arguments (for example: `-Dprop=key`), and another thing is the program arguments which is addressed with `quarkus.kubernetes.arguments`.